library/src/main/camera2/com/google/android/cameraview/PictureCaptureCallback.java

```diff
@@ -0,0 +1,105 @@
+package com.google.android.cameraview;
+
+import android.hardware.camera2.CameraCaptureSession;
+import android.hardware.camera2.CaptureRequest;
+import android.hardware.camera2.CaptureResult;
+import android.hardware.camera2.TotalCaptureResult;
+import android.support.annotation.NonNull;
+
+/**
+ * A {@link CameraCaptureSession.CaptureCallback} for capturing a still picture.
+ */
+abstract class PictureCaptureCallback extends CameraCaptureSession.CaptureCallback {
+    private static final int MAX_LOCK_FOCUS_ATTEMPTS = 3;
+
+    public static final int STATE_PREVIEW = 0;
+    public static final int STATE_LOCKING = 1;
+    public static final int STATE_LOCKED = 2;
+    public static final int STATE_PRECAPTURE = 3;
+    public static final int STATE_WAITING = 4;
+    public static final int STATE_CAPTURING = 5;
+
+    private int mState;
+    private int mLockFocusAttemptsCount;
+
+
+    public void setState(int state) {
+        mState = state;
+    }
+
+    public void clearLockFocusAttemptsCount() {
+        mLockFocusAttemptsCount = 0;
+    }
+
+    @Override
+    public void onCaptureProgressed(@NonNull CameraCaptureSession session,
+                                    @NonNull CaptureRequest request,
+                                    @NonNull CaptureResult partialResult) {
+        process(partialResult);
+    }
+
+    @Override
+    public void onCaptureCompleted(@NonNull CameraCaptureSession session,
+                                   @NonNull CaptureRequest request,
+                                   @NonNull TotalCaptureResult result) {
+        process(result);
+    }
+
+    private void process(@NonNull CaptureResult result) {
+        switch (mState) {
+            case STATE_LOCKING: {
+                Integer af = result.get(CaptureResult.CONTROL_AF_STATE);
+                if (af == null) {
+                    break;
+                }
+                Integer ae = result.get(CaptureResult.CONTROL_AE_STATE);
```

Comment:
Most of this code was already there, it was jut copied. Cameraview and the camera API use AF and AE everywhere to refer to Auto Focus and Auto Exposure.